plugins/alerts/frontend/public/javascripts/countly.views.js

```diff
@@ -12,811 +12,483 @@
-            self.lastAlertAppId = $('#' + alertID).attr('app-id');
-            var formData = alertsPlugin.getAlert(alertID);
-            self.widgetDrawer.loadData(formData);
-            $(self.widgetDrawer.drawer).find('.title span').first().html(jQuery.i18n.map["alert.Edit_Your_Alert"]);
-
-            self.widgetDrawer.drawer.open();
-            $(self.widgetDrawer.drawer).addClass("open editing");
-        });
-
     },
-    renderCommon: function(refresh) {
-        var self = this;
-        $(this.el).html(this.template({"email-placeholder": jQuery.i18n.map["alert.email-place-holder"]}));
-        if (!refresh) {
-            var views = ["starView", "crashesView"];
-            views.forEach(function(view) {
-                if (!app[view]) {
-                    CountlyHelpers.notify({
-                        clearAll: false,
-                        type: 'warning',
-                        title: jQuery.i18n.map["alert." + view + "-disabled-title"],
-                        message: jQuery.i18n.map["alert." + view + "-disabled-desc"],
-                        info: jQuery.i18n.map["alert." + view + "-disabled-suggest"],
-                        delay: 5000,
-                        sticky: false
-                    });
-                }
-            });
-        }
-        if (countlyAuth.validateCreate(self.featureName)) {
-            $('#create-alert').show();
+    watch: {
+    },
+    props: {
+        controls: {
+            type: Object
         }
-        this.renderTable();
-        this.prepareDrawer();
     },
-    updateCount: function() {
-        var count = alertsPlugin.getCount();
-        $("#alerts-running-sum").text(count.r);
-        $("#alerts-total-sum").text(count.t);
-        $("#alerts-today-sum").text(count.today);
+    mounted: function () {
     },
-    widgetDrawer: {
-        loadRatingOptions: function(selected) {
-            var ratings = [
-                {value: 1, name: jQuery.i18n.map["star.one-star"]},
-                {value: 2, name: jQuery.i18n.map["star.two-star"]},
-                {value: 3, name: jQuery.i18n.map["star.three-star"]},
-                {value: 4, name: jQuery.i18n.map["star.four-star"]},
-                {value: 5, name: jQuery.i18n.map["star.five-star"]},
-            ];
-            $("#single-target2-dropdown").clySelectSetItems(ratings);
-            if (selected) {
-                $("#single-target2-dropdown").clySelectSetSelection(ratings[selected - 1].value, ratings[selected - 1].name);
+    methods: {
+        appSelected: function (val, notReset) {
+            var self = this;
+            this.$data.apps= [val];
+            if (val) {
+                countlyEvent.getEventsForApps([val], function(eventData) {
+                    const eventOptions = eventData.map(function(e){
+                        return {value: e.value, label: e.name};
+                    });
+                    self.$data.alertDefine['event']["target"] = eventOptions;
```

Comment:
instead of **self.$data.alertDefine** you could have used **this.alertDefine** since this has the scope of the current component.